### PR TITLE
Adding support for install AzCLI in standard agent image

### PR DESCRIPTION
**Pull request by sachinma**

### CODE REVIEW

**Comment by stepro** [comment]
Can you describe the scenarios that use the az CLI? Are there VSTS tasks that depend on it or are you looking to enable calling az from a generic task (like run script)?

**Comment by sachinma** [comment]
Yes, we do have a built-in task "https://github.com/Microsoft/vsts-tasks/tree/master/Tasks/AzureCLI" which lets you run az based scripts. Az CLI is present in VS 2017 for quite some time. We have had some customers looking for it in Hosted Linux Pool - hence this PR.

File: README.md [hunk 1]
```diff
@@ -99,6 +99,7 @@ These derived images include a set of standard capabilities that enable many of
 - Python 2.7.12 and Python 3.5.2
 - Node.js 8.4.0
 - .NET Core SDK 2.0.0
+- Az CLI 2.0
```
Comment (stepro) [inline README.md]: Can we use the full name "Azure CLI 2.0" here?

File: ubuntu/standard/Dockerfile.template [hunk 2]
```diff
@@ -80,6 +80,15 @@ RUN curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /et
 # Run dotnet command to populate initial cache
 RUN dotnet help
 
+# Configure AzCLI 2.0  Instructions taken from https://docs.microsoft.com/en-us/cli/azure/install-azure-cli
+RUN echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ wheezy main" | \
+     tee /etc/apt/sources.list.d/azure-cli.list \
```
Comment (stepro) [inline ubuntu/standard/Dockerfile.template]: Do we know that "wheezy" as a debian version identifier is going to be correct across potentially multiple Ubuntu versions (16.04 only for now, but may be newer versions in the future)? The .NET Core SDK installation instructions expect you to pass an Ubuntu version identifier instead (see line 74).

Comment (stepro) [reply ubuntu/standard/Dockerfile.template]: Also, please line up the indentation of the "echo" and "tee" commands.

File: ubuntu/standard/Dockerfile.template [hunk 3]
```diff
@@ -80,6 +80,15 @@ RUN curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /et
...
+     && apt-key adv --keyserver packages.microsoft.com --recv-keys 417A0893 \
+     && apt-get install -y --no-install-recommends apt-transport-https \
```
Comment (stepro) [inline ubuntu/standard/Dockerfile.template]: Please follow the formatting for multi-line RUN commands by adding only a single character before the "&&" (this lines up the commands being executed in sequence for easier readability).

File: ubuntu/standard/Dockerfile.template [hunk 4]
```diff
@@ -80,6 +80,15 @@ RUN curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /et
...
+     && apt-get update && apt-get install -y --no-install-recommends azure-cli \
```
Comment (stepro) [inline ubuntu/standard/Dockerfile.template]: Place these two commands on separate lines.

Comment (sachinma) [reply ubuntu/standard/Dockerfile.template]: done

File: ubuntu/standard/Dockerfile.template (hunk 3 above)
Comment (sachinma) [reply ubuntu/standard/Dockerfile.template]: done

File: README.md (hunk 1 above)
Comment (sachinma) [reply README.md]: done

File: ubuntu/standard/Dockerfile.template (hunk 2 above)
Comment (sachinma) [reply ubuntu/standard/Dockerfile.template]: Currently only the wheezy version identifier is available with azure-cli source. In their documentation, they are not specifying any version specific install steps. I tried to run this on Ubuntu 17.04 and it works - but I am not sure if they will keep this up-to-date with future Ubuntu versions. I think we might have to work with them in such cases when we add support for more Ubuntu versions. I am not sure that we should ask AzCli team to create different version identifiers in their source even when the same step can work today.

**Comment by stepro** [comment]
Looks good. I'll merge and build new images and push them to Docker Hub.

**Comment by stepro** [comment]
The images have been updated on Docker Hub.

**Comment by sachinma** [comment]
Thanks Stephen.

**Pull request merged**